File: SYP.Web/Modules/ServerTypes/Catalog/ProductPackingForm.ts
import { BooleanEditor, initFormType, IntegerEditor, PrefixedContext, StringEditor, TextAreaEditor } from "@serenity-is/corelib";

export interface ProductPackingForm {
    Code: StringEditor;
    Name: StringEditor;
    Description: TextAreaEditor;
    Quantity: IntegerEditor;
    IsActive: BooleanEditor;
}

export class ProductPackingForm extends PrefixedContext {
    static readonly formKey = 'Catalog.ProductPacking';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!ProductPackingForm.init) {
            ProductPackingForm.init = true;

            var w0 = StringEditor;
            var w1 = TextAreaEditor;
            var w2 = IntegerEditor;
            var w3 = BooleanEditor;

            initFormType(ProductPackingForm, [
                'Code', w0,
                'Name', w0,
                'Description', w1,
                'Quantity', w2,
                'IsActive', w3
            ]);
        }
    }
}

File: SYP.Web/Modules/Catalog/ProductPacking/ProductPackingForm.cs
using Serenity.ComponentModel;

namespace SYP.Catalog.Forms;

[FormScript("Catalog.ProductPacking")]
[BasedOnRow(typeof(ProductPackingRow), CheckNames = true)]
public class ProductPackingForm
{
    public string Code { get; set; }
    public string Name { get; set; }

    [TextAreaEditor(Rows = 3)]
    public string Description { get; set; }

    public int Quantity { get; set; }

    [DefaultValue(true)]
    public bool IsActive { get; set; }
}
